lvmsnapshotter/lvmsnapshotter.go

```diff
@@ -47,30 +48,32 @@ func NewSnapshotter(vgname string, lvpoolname string) (snapshots.Snapshotter, er
 
 	// Check if the mount directory exists. If not, create it. if it is a file then exit
 
-	if _, err := checkVG(vgname); err != nil {
+	if _, err = checkVG(vgname); err != nil {
 		return nil, errors.Wrap(err, "VG not found")
 	}
 
-	_, err := checkLV(vgname, lvpoolname)
+	_, err = checkLV(vgname, lvpoolname)
 	if err != nil {
 		return nil, errors.Wrap(err, "LV not found")
 	}
 
 	_, err = checkLV(vgname, metavolume)
 	if err != nil {
 		// Create a volume to hold the metadata.db file.
-		if _, err := createLVMVolume(metavolume, vgname, lvpoolname, "", snapshots.KindUnknown); err != nil {
+		if _, err = createLVMVolume(metavolume, vgname, lvpoolname, "", snapshots.KindUnknown); err != nil {
 			llog.Printf("Unable to create the metadata volume\n")
 			return nil, err
 		}
 	} else {
 		llog.Printf("Re-using the existing volume\n")
 	}
-	if _, err := toggleactivateLV(vgname, metavolume, true); err != nil {
+	if _, err = toggleactivateLV(vgname, metavolume, true); err != nil {
 		return nil, errors.Wrap(err, "Unable to activate metavolume")
 	}
-	if fi, err := os.Stat(metaVolumeMountPath); os.IsNotExist(err) {
+	var fi os.FileInfo
```

Comment:
Can i be pedantic and ask you to move all the variable definitions to the top of the function please?